### PR TITLE
[ADD] mass_mailing_confirm addon

### PR DESCRIPTION
**Pull request by antespi**
This module extends the functionality of mass_mailing addon adding a
confirmation dialog after user click to "Send to all" button at Mass mailing
form view, in order to confirm that operation.

It is a very simple addon but save from human error clicking "Send to all" button by accident.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
As I said directly in the chat: this can be done with confirm attribute of the button.

**Comment by antespi** [comment]
Thanks @pedrobaeza for your sincerity. As my grandfather said:

```
You'll never go to sleep without learning another lesson 
```

**Comment by pedrobaeza** [comment]
Sorry, I meant another thing. I have rephrased the sentence to the meaning I intend.

**Comment by antespi** [comment]
squashed
Maybe we can propuse this to be fixed at odoo/odoo, isn't it?

File: mass_mailing_confirm/views/mass_mailing_view.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+<data>
+
+<record id="view_mail_mass_mailing_popup_form" model="ir.ui.view">
+    <field name="name">mass.mailing.form</field>
+    <field name="model">mail.mass_mailing</field>
+    <field name="priority">90</field>
+    <field name="inherit_id" ref="mass_mailing.view_mail_mass_mailing_form"/>
+    <field name="arch" type="xml">
+
+        <xpath expr="//header/button[4]" position="attributes">
```
Comment (hbrunn) [inline mass_mailing_confirm/views/mass_mailing_view.xml:14]: I think it's safer to select the button by its name, not its position

File: mass_mailing_confirm/views/mass_mailing_view.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            <attribute name="confirm">Are you sure?</attribute>
+        </xpath>
+        <xpath expr="//header/button[3]" position="attributes">
```
Comment (hbrunn) [inline mass_mailing_confirm/views/mass_mailing_view.xml:17]: see above

File: mass_mailing_confirm/views/mass_mailing_view.xml (hunk 1 above)
Comment (antespi) [inline mass_mailing_confirm/views/mass_mailing_view.xml:14]: Sorry @hbrunn I don't know how to do better. This is the original inherited view:

```
        <record model="ir.ui.view" id="view_mail_mass_mailing_form">
            <field name="name">mail.mass_mailing.form</field>
            <field name="model">mail.mass_mailing</field>
            <field name="arch" type="xml">
                <form string="Mass Mailing">
                    <header>
                        <button name="action_test_mailing" type="object"
                            class="oe_highlight" string="Test Mailing"
                            attrs="{'invisible': ['|', ('state', '!=', 'draft'), ('body_html', '=', False)]}"/>
                        <button name="action_test_mailing" type="object"
                            attrs="{'invisible': ['&amp;', ('state', '=', 'draft'), ('body_html', '!=', False)]}"
                            string="Test Mailing"/>
                        <button name="send_mail" type="object"
                            attrs="{'invisible': ['|', ('state', '=', 'done'), ('body_html', '=', False)]}"
                            class="oe_highlight" string="Send to All"/>
                        <button name="send_mail" type="object"
                            attrs="{'invisible': ['&amp;', ('state', '!=', 'done'), ('body_html', '!=', False)]}"
                            string="Send to All"/>
                        <field name="state" widget="statusbar"/>
                    </header>
                    <div class="oe_form_box_info oe_text_center" attrs="{'invisible': [('scheduled', '=', 0)]}">
                        <p><strong>
                            <field name="scheduled" class="oe_inline"/>
                            emails are in queue and will be sent soon.
                        </strong></p>
                    </div>
                    <sheet>
(...)
```

As you can see there are two bottons, one with class `oe_highlight` and other without it. So I can't match by name, isn't it?

**Comment by antespi** [comment]
BTW, I've just PRed it to odoo/odoo:8.0 https://github.com/odoo/odoo/pull/9796
I think this fix can be merged in mainstream.

File: mass_mailing_confirm/README.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
Comment (elicoidal) [inline mass_mailing_confirm/README.rst:57]: https

File: mass_mailing_confirm/i18n/es.po [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+"Language-Team: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
+"Plural-Forms: \n"
+
+#. module: mass_mailing_confirm
+#: view:mail.mass_mailing:mass_mailing_confirm.view_mail_mass_mailing_popup_form
+msgid "Are you sure?"
+msgstr "¿Estás seguro?"
```
Comment (elicoidal) [inline mass_mailing_confirm/i18n/es.po:21]: "¿Está seguro?"

File: mass_mailing_confirm/README.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+====================
+Mass mailing confirm
+====================
+
+This module extends the functionality of mass_mailing addon adding a
+confirmation dialog after user click to "Send to all" button at Mass mailing
```
Comment (pedrobaeza) [inline mass_mailing_confirm/README.rst:10]: s/to/in

File: mass_mailing_confirm/views/mass_mailing_view.xml (hunk 1 above)
Comment (pedrobaeza) [inline mass_mailing_confirm/views/mass_mailing_view.xml:14]: You can restrict a bit more:

```
<xpath expr="//button[@name='action_test_mailing'][0]" position="attributes">
...
<xpath expr="//button[@name='action_test_mailing'][1]" position="attributes">
```

File: mass_mailing_confirm/README.rst (hunk 5 above)
Comment (elicoidal) [inline mass_mailing_confirm/README.rst:10]: are you sure?
"confirmation dialog after user click "Send to all" button in Mass mailing"

**Comment by rafaelbn** [comment]
:+1: 



Waiting for an answer upstream https://github.com/odoo/odoo/pull/9796

**Comment by antespi** [comment]
Odoo has merged this https://github.com/odoo/odoo/pull/9796 :smile:
So, I close this PR. 
Thanks all anyway

**Pull request closed without merging**